lib/modules/dosomething/dosomething_api/includes/Transformer.php

```diff
@@ -217,16 +217,123 @@ protected function transformCollection($items, $method = 'transform') {
+   *   - secondary_cause:
    *
    * @return array
    */
   protected function transformCampaign($data) {
-    return array(
-      'campaign' => array(
-        'id' => $data->nid,
-        'title' => $data->title,
-      ),
+    $output = array(
+      'id' => $data->id,
+      'title' => $data->title,
     );
+
+    if ($data instanceof Campaign) {
+      $output['tagline'] = $data->tagline;
+      $output['created_at'] = $data->created_at;
+      $output['updated_at'] = $data->updated_at;
+      $output['time_commitment'] = $data->time_commitment;
+      // $output['type'] = $data->type; //@TODO: Should type be included? Consider there is an SMS Campaign type...
+
+      if ($data->status) {
+        // @TODO: Should the status default to "active"?
+        $output['status'] = $data->status;
+      }
+
+      if ($data->cover_image) {
+        foreach ($data->cover_image as $key => $image) {
+          if (!is_null($image)) {
+            $output['cover_image'][$key] = $this->transformMedia($image, 'square');
+          }
+        }
+      }
+
+      if ($data->facts['problem']) {
+        $output['facts']['problem'] = $data->facts['problem']['fact'];
+      }
+
+      if ($data->facts['solution']) {
+        $output['facts']['solution'] = $data->facts['solution']['fact'];
+      }
+
+      if ($data->facts['sources']) {
+        $output['facts']['sources'] = $data->facts['sources'];
+      }
+
+      if ($data->solutions['copy']) {
+        $output['solutions']['copy'] = $data->solutions['copy'];
+      }
+
+      if ($data->solutions['support_copy']) {
+        $output['solutions']['support_copy'] = $data->solutions['support_copy'];
+      }
+
+      if ($data->causes['primary']) {
+        $output['causes']['primary'] = $data->causes['primary'];
+      }
+
+      if ($data->causes['secondary']) {
+        $output['causes']['secondary'] = $data->causes['secondary'];
+      }
+
+      if ($data->action_types['primary']) {
+        $output['action_types']['primary'] = $data->action_types['primary'];
+      }
+
+      if ($data->action_types['secondary']) {
+        $output['action_types']['secondary'] = $data->action_types['secondary'];
+      }
+
+      if ($data->issue) {
+        $output['issue'] = $data->issue;
+      }
+
+      if ($data->tags) {
+        $output['tags'] = $data->tags;
+      }
+
+    }
+
+    return $output;
+  }
+
+
+
+  protected function transformMedia($data, $aspect_ratio = NULL) {
+    if ($data['type'] === 'image') {
+      $output = array();
+
+      $output['uri'] = $data['sizes'][$aspect_ratio]['uri'];
+
+      foreach ($data['sizes'] as $key => $size) {
+        if (isset($size['uri'])) {
+          $output['sizes'][$key]['uri'] = $size['uri'];
+        }
+      }
+
+      $output['type'] = $data['type'];
+      $output['dark_background'] = $data['dark_background'];
+
+      return $output;
+    }
+
+    if ($data['type'] === 'video') {
+      // @TODO: include video transformation code here!
```

Comment:
Might be worth creating a GH issue for this too, although the only current use we'd have for it is just displaying the video on the web using this class instead of the current implementation.